dlf/plugins/pageview/tx_dlf_utils.js

```diff
@@ -561,38 +561,48 @@ dlfUtils.isCorsEnabled = function(imageObjs) {
     // fetch data from server
     // with access control allowed
     let response = true;
-
+    const type = 'HEAD';
     imageObjs.forEach(function(imageObj) {
         let url = imageObj.mimetype === dlfUtils.CUSTOM_MIMETYPE.ZOOMIFY
-            ? imageObj.url.replace('ImageProperties.xml', 'TileGroup0/0-0-0.jpg')
-            :
-            imageObj.mimetype === dlfUtils.CUSTOM_MIMETYPE.IIIF
-                ? dlfViewerSource.IIIF.getMetdadataURL(imageObj.url)
+            ? dlfUtils.buildXhrURI(imageObj.url.replace('ImageProperties.xml', 'TileGroup0/0-0-0.jpg'), true)
+            : imageObj.mimetype === dlfUtils.CUSTOM_MIMETYPE.IIIF
+                ? dlfUtils.buildXhrURI(dlfViewerSource.IIIF.getMetdadataURL(imageObj.url), false)
```

Comment:
I would never assume that all IIIF servers are properly configured, but if they are not, then it should not break the client.  If `isCorsEnabled` preflight fails, then the client should simply throw an exception and move on.  I would not say that you _should not_ try to correct a misconfigured server with a response header rewriting, because this depends on the scope of your domain, but doing so introduces stateful dependencies, and a potentially a lot of extra complexity (and performance issues).  

A server (in a proper RESTful implementation of CORS), _should_ provide a preflight `OPTIONS` response that includes the Access-Control-Allow-Origin header, but this is not always done (and is not done with IIPsrv).  I think that the `isCorsEnabled` function with a `HEAD` request that does the preflight is all that is required to confirm that a server supports CORS for this client before the `GET` for that resource is sent. 

I think that grouping resources by server/path for the benefit of avoiding individual preflight requests may have benefit if the client was loading many (hundreds) of images on the same page, but this is not done with this client AFAIK.  

Regarding `mixed content` and HTTPS, I am not clear on what the exact problem is.  I have tested running TYPO3 on HTTPS and getting images from HTTP and this just throws warnings in the browser, but nothing fatal.  I think that mixed content is unavoidable, unless you blacklist HTTP resources when you initially parse the manifest.